tracer/src/Datadog.Trace/SpanExtensions.cs

```diff
@@ -62,5 +77,29 @@ public static void SetUser(this ISpan span, UserDetails userDetails)
                 localRootSpan.SetTag(Tags.User.Scope, userDetails.Scope);
             }
         }
+
+        private static void SetPropagatedId(ISpan span, UserDetails userDetails, ISpan localRootSpan, TraceContext traceContext)
+        {
+                var propagationHeaderMaxLength =
+                    traceContext?.Tracer.Settings.TagPropagationHeaderMaxLength ??
+                        TagPropagation.OutgoingPropagationHeaderMaxLength;
+
+                if (userDetails.Id.Length > propagationHeaderMaxLength)
+                {
+                    Log.Warning<string, int>("{Id} is {IdLength} bytes long, which is longer than the configured max length of {MaxLength}", userDetails.Id, userDetails.Id.Length, propagationHeaderMaxLength);
+                    return;
+                }
+
+                var base64UserId = Convert.ToBase64String(Encoding.UTF8.GetBytes(userDetails.Id));
+
+                if (base64UserId.Length > propagationHeaderMaxLength)
+                {
+                    Log.Warning<string, int>("base64 encoded {Id} is {IdLength} bytes long, which is longer than the configured max length of {MaxLength}", base64UserId, base64UserId.Length, propagationHeaderMaxLength);
+                    return;
+                }
+
+                const string propagatedUserIdTag = TagPropagation.PropagatedTagPrefix + Tags.User.Id;
+                localRootSpan.SetTag(propagatedUserIdTag, base64UserId);
```

Comment:
Instead of this, use
```csharp
traceContext.Tags.SetTag(...);
```
This adds the tag to the trace tags collection, and takes care of sending it to the agent according to the endpoint schema (e.g. adding it to root spans for endpoint 0.4) and injecting it into downstream headers.